feat: add unit tests

thehilll opened the pull request:
Plan now is to start w/your initial branch and step through to get to the full version I have tested.  Initial PR adds only unit tests.

redstreet commented:
Sounds good! Let me take a look later today.

This one is not recognized:
`EXCHANGED TO SPAXX FIDELITY GOVERNMENT MONEY MARKET (SPAXX) (Cash)`

redstreet commented:
Yes, correct, it's in my own files. I'm just dumping what I find here as I find them so we can address either in this PR or subsequent PRs.

thehilll commented:
> Because `self.transaction_type_map` is all uppercase and some transactions come in as title case we should either match those title case transactions exactly (e.g. revert [3ad9043](https://github.com/redstreet/beancount_reds_importers/commit/3ad9043ed49aa11f73f6f02f789ddaf78c36262e)) or normalize the data by adding something like `rdr = rdr.convert("type", 'upper')` to `prepare_table()`.

I've added this to both branches in my fork.  Should show up here as we progress through PRs.  If the decision is to handle this in `csvreader` then it is easy to back out.

> We also need to change "WITHDRAWAL" to "WITHDRAWALS" in the self.transaction_type_map. (I noted this in PR #173, but repeating here for completeness)

I've added this as well.

patbakdev commented:
It would also be nice if the the *Transaction* Narration field contained the *CSV* Description field in cases where the *Transaction* Symbol field was empty/missing so that we end up with something like:

```
2025-12-11 * "Exchange Out" "[NOTICKER] VANG 500 IDX IS SEL"
  todo: "TODO: this entry is incomplete until lots are selected (bean-doctor context <filename> <lineno>)"
  todo: "TODO: No ticker was specified in the input file.
  Assets:ACCOUNT:NOTICKER                              -13.994 NOTICKER {}
  Income:ACCOUNT:NOTICKER
  Assets:ACCOUNT:Cash                                  -12345.82 USD
```

Let me know if you would rather me open new/separate issues for each of these.

thehilll commented:
> I think we are still missing the commit for:
> 
> > Because self.transaction_type_map is all uppercase and some transactions come in as title case we should either match those title case transactions exactly (e.g. revert [3ad9043](https://github.com/redstreet/beancount_reds_importers/commit/3ad9043ed49aa11f73f6f02f789ddaf78c36262e)) or normalize the data by adding something like rdr = rdr.convert("type", 'upper') to prepare_table().
> 

I do have a fix for this staged…was just waiting to do PR so we didn’t get a big pile of them.  I can include this next.

redstreet commented:
Thank you @patbakdev. Yes, separate issues for all the pending ones would be great, just to get organized.

And thank you @thehilll, I'll look forward to your next PR!
